### ISSUE TITLE
Create NULL_PARAMETER failure reason

### ISSUE THREAD

**Issue by MrNemo64** [lokka30/Treasury#54]
As said by @MrIvanPlays [here](https://github.com/lokka30/Treasury/pull/51#discussion_r762958881), just by adding the `@NotNull` anotation we don't ensure that a parameter won't be null
> the plugin is going to be build and deployed to maven repositories via a CI and not by intellij/human, so if you want that code to generate not just if you compile via intellij, you need to find a maven plugin which does it and put it. But that has nothing to do with this PR.

This means that a null parameter could be provided when non null parameters were expected, in this case the EconomySubscriber should failt but the only FailureReason that could be provided is `OTHER_FAILURE`, as there is no failure reason for null parameters
This is why I propose to add a `NULL_PARAMETER` failure reason
The other option would be to find a maven plugin that allows us to have the `@NotNull` anotation actually check the parameter but in my opinion, we should avoid using others plugins when necessary

**Comment by lokka30** [comment]
I belive we should add the failure reason instead of adding more Maven plugins.

**Comment by MrIvanPlays** [comment]
Maven plugin is the best option. That way we're not bloating the API.

**Comment by lokka30** [comment]
So long the maven plugin is very stable and does its job properly I don't mind using that.

**Comment by MrIvanPlays** [comment]
I did some more research and there's no maven plugin for such thing. Neither InteliJ parses the annotations as jikoo mentioned as I also tried compiling with maven through intellij.

Or my decompiler is broken... I decompiled spigot api and they're staying there too. So these annotations are just code sugar after the debate we had. If we really want something which enforces non null via an annotation we shall use lombok, but as you know lombok is a shady business and will push away many potential contributors (~~such as me~~). 

The 3 options we have:
- Make it stay as is
- NULL_PARAMETER failure reason
- Remove the annotations
